app/Listeners/LoginListener.php

```diff
@@ -25,10 +27,13 @@ public function __construct()
      */
     public function handle($event)
     {
+        $userIp = $request->ip();
+        $locationData = \Location::get($userIp);
```

Comment:
Let's add a `use` statement instead the fully qualified class name `\Location` (see [this issue](https://github.com/eclectic-bytech/webcourses-ngx-v4/issues/192) for an example in the final comment's linked commit.) Also checkout [this 5 minute tutorial on name spaces](https://symfonycasts.com/screencast/php-namespaces/namespaces) that explains this implementation. 

The [official docs](https://packagist.org/packages/stevebauman/location) recommend putting an if statement around the Location::Get. This won't work inside `$event->user->update`. To get around that, we can set the country code to `XYZ` and then update it with a real one if found. This way there's a value for the update statement, even if the GeoID fails for any reason.